src/mcode-pilot/utils/filterTreatmentData.js

```diff
@@ -100,14 +101,13 @@ function isSimilarPatient(treatmentDataPatient, fOptions) {
             }
         } else if (filter.mcodeElement === 'shr.core.DateOfDiagnosis') {
             // age at diagnosis
-            const [birthYear] = birthDate.split('-').map((value) => parseInt(value, 10));
-            const [dxYear] = diseaseStatus.diagnosisDate.split('-').map((value) => parseInt(value, 10));
-            const dxAge = dxYear - birthYear;
-
+            const birthYear = new Date(birthDate).getTime();
+            const dxYear = new Date(diseaseStatus.diagnosisDate).getTime();
+            const dxAge = (dxYear - birthYear)/31557600000; // 1000*60*60*24*365.25
             if (dxAge < minValue || dxAge > maxValue) {
                 return false;
             }
-        } else if (filter.mcodeElement === 'shr.core.Race' && value !== _.lowerCase(race)) {
+        } else if (filter.mcodeElement === 'shr.core.Race' && value !== race.toLowerCase()) {
             return false;
         } else if (filter.mcodeElement === 'shr.core.BirthSex' && value !== _.lowerCase(gender)) {
```

Comment:
I think I just forgot to include all the `lowerCase`s.  

Also I forgot to change the birth year calc for both date of diagnosis and date of birth because I was using the two separate methods to check speed increase.  I will sync everything to use one style.